### DISCUSSION TITLE
Support for 4-bit PCX images

### DISCUSSION THREAD

**Discussion by Tjossul** [ImageMagick/ImageMagick#6689] [Contribute]
### Is your feature request related to a problem? Please describe.

For some reason I need to send 4-bit PCX images to a (n old) printer because it can only handle that format.

### Describe the solution you'd like

The library supports 8-bit PCX and above. It would be cool if PCX images could also be saved in 4-bit format.

**Comment by sijnstra** [comment]
I would also like to see this feature. Currently the only ways I can reliably save a 4 bit per plane, single plane .PCX image file is using irfanview or GIMP, rather than a command line tool. I would love to have this scriptable through imagemagick.

**Comment by rnieblesealo** [comment]
I'd like to tackle this!

After taking a look at `coders/pcx.c`, it looks like ImageMagick is only able to write PCX output as either 1 bit * 1 plane ( monochrome ) or 8 bits * ( 1 | 3 | 4 ) planes ( 1 plane for 8-bit indexed color, 3 planes for RGB, and 4 planes for RGBA ).

To implement 4 bit PCX ( for 16 bit indexed color ), it seems we need to add a third path that's able to write 1 bit * 4 planes.

My approach:

1. Check the image to set the PCX plane and bit counts for this mode (`pcx_info.planes` and `pcx_info.bits_per_pixel` ), adjusting ImageMagick's internal representation accordingly ( `image->type = PaletteType` )
2. Writing an additional path that writes pixels according to these settings ( Still working on understanding this! )

@urban-warrior Any comments? This'd be my first contribution to ImageMagick ( and open source in general! ) so any help is greatly appreciated!

**Comment by urban-warrior** [comment]
This project welcomes contributions that enhance ImageMagick’s capabilities. Submit your patches as pull requests and include unit tests that demonstrate the patch works correctly and is robust. Thank you in advance for your contribution.
